# 06 - Recap

Quick recap of the pipeline: I started with 139 companies from the robotics supply chain, and I ended up with 58 tickers in the final PEG vs ROIC chart. This notebook shows where the other companies were dropped.

In [8]:
import pandas as pd
import plotly.express as px

For each step, I just count the rows in the file produced by that step.

In [9]:
initial_dataset = len(pd.read_csv("../../data/raw/stocks/stocks.csv"))
public_and_cleaned = len(pd.read_csv("../../data/processed/stocks/stocks_cleaned.csv"))
tickers_verified = len(pd.read_csv("../../data/processed/stocks/stocks_transformed_verified.csv"))
with_kpis = len(pd.read_csv("../../data/processed/stocks/stocks_kpis.csv"))

print(f"Initial dataset: {initial_dataset}")
print(f"Public and cleaned: {public_and_cleaned}")
print(f"Tickers verified: {tickers_verified}")
print(f"With KPIs: {with_kpis}")

Initial dataset: 139
Public and cleaned: 102
Tickers verified: 100
With KPIs: 58


In [10]:
# Same filters as the scatter plot
funnel = pd.DataFrame({
    "stage": ["Initial dataset", "Public and cleaned", "Tickers verified", "With KPIs"],
    "count": [initial_dataset, public_and_cleaned, tickers_verified, with_kpis],
})

funnel

,stage,count
0,Initial dataset,139
1,Public and cleaned,102
2,Tickers verified,100
3,With KPIs,58


Plot the result as a simple bar chart, one bar per step.

In [11]:
fig = px.bar(
    funnel,
    x="stage",
    y="count",
    text="count",
    color="stage",
    title="Screening funnel",
)

fig.update_traces(textposition="outside", showlegend=False)
fig.update_layout(xaxis_title="", yaxis_title="Companies", showlegend=False)

fig.show()

Details of the cleaning and transformations:

* **139 to 102 (-37):** removed the private companies and non-profits, then merged the companies sharing one ticker (a subsidiary quoted with its parent's ticker: `GOOGL`, `RRX`, `005380`), so no duplicate ticker survives.
* **102 to 100 (-2):** only `SHA` and `MOGA` could not be found on yfinance.
* **100 to 58 (-42):** removed tickers missing the PEG and / or the ROIC + the negative PEG ratios (a negative PEG does not mean cheap) and the outliers above PEG = 10.